docs/reference/configuration.md

```diff
@@ -0,0 +1,195 @@
+# Configuration
+
+When defining a configuration file for Screwdriver, we wanted to follow some basic principles:
+
+ - Strive to be terse but legible
+ - Can translate into a simple shell script
+ - Use familiar concepts (environment variables and shell commands)
+ - Reproducible locally (in/out of containers)
+ - Should promote continuous delivery best practices without limiting experimentation
+ - Extensible for future features
+
+## Inspiration
+
+Screwdriver's configuration itself has changed significantly over the past 3 years.  For each iteration, we took a look at what was working well and what still needed improvement.  This latest format is inspired from a combination of the following tools:
+
+ - [Wercker](http://devcenter.wercker.com/docs/wercker-yml/index.html)
+ - [Travis](https://docs.travis-ci.com/user/getting-started/)
+ - [Drone.io](http://readme.drone.io/usage/overview/)
+
+## Format
+
+We chose to use standard YAML formatting as it provided:
+
+ - a clean interface
+ - human readable
+ - loose formatting requirements
+ - allows inline documentation
+ - easily checked for syntax errors
+
+The number one downside is the lack of standard schema validation tools.  But we considered the positives to outweigh the ease of building a validation tool.
+
+## Basic Example
+
+```yaml
+
+rebuild_on:
+    - git@github.com:screwdriver-cd/plugin-jobs
+    - git@github.com:screwdriver-cd/plugin-builds
+
+shared:
+    image: node:$NODE_VERSION
+
+    matrix:
+        NODE_VERSION: [4,5,6]
+
+jobs:
+    pr:
+        steps:
+            init: npm install
+            test: npm test
+
+    main:
+        steps:
+            init: npm install
+            test: npm test
+            publish: npm publish
+```
+
+## Global Features
+
+These are specified at the root of the `screwdriver.yaml` and apply to the entire pipeline.
+
+### Workflow
+
+> I want to easily describe the flow of my pipeline, so that it is clear what jobs will run in what order.
+
+In the past we had users specifying `triggers` in each job - which defined what jobs were started on the completion of that job.  This was disjointed and did not provide an easy overview of the flow of the pipeline.  Because of that, we added a basic `order` feature - an array of jobs to execute in that order.  This worked for most simple pipelines, but was unusable when users wanted to do parallelization or detached jobs.  Workflow is the next iteration of this, a clean representation of the flow of a pipeline.
```

Comment:
Do we want to include all this? I'm reading this from a **user's point of view** and this seems like a lot of unnecessary information. I don't care too much about how screwdriver used to be, or why the team decided to make certain decisions. I only care about how screwdriver works **now** and how to use it, especially since this is the configuration page. Maybe it's more appropriate to include our design decision on a separate page?